### PR TITLE
Update documentation to rationalize the terminology

### LINKED ISSUE #75: Clarify vocabulary
The vocabulary currently used by DM is extremely confusing:

* Configuration files that contain no substitution points are called templates.
* Configuration files that do contain substitution points are called types, or more precisely `template types`.
* Python or Jinja files that define abstract types are called templates.
* Kubernetes primitives like Pod, Service and Replication Controller, are called configurable resource types. There are no other uses of this term, unlike in DM for GCP.

On top of this confusion, the term `type` is opaque. Users looking for simple templates might not think to look at types. Everyone knows what a template is. It's a piece of content that contains substitution points. However, a type could be just about anything.

We need to clean up the terminology, focusing more on making it intuitive to the Kubernetes community than on following the precedents set by DM for GCP.

### PR DESCRIPTION
**Pull request by jackgr**
This PR partially fixes #75, by updating the documentation. If we agree with these changes, then we'll need another PR to make a few small changes to `dm` usage, the top level directory name for the type registry in this repo, etc.

Specifically, this PR establishes the following terminology:
- Configuration files that contain no substitution points are called configurations.
- Python or Jinja files that do contain substitution points are called templates.
- Kubernetes primitives like Pod, Service and Replication Controller, are called primitives.
- Both templates and primitives are types.

### CODE REVIEW

File: docs/design/design.md [hunk 1]
```diff
@@ -318,58 +368,67 @@ instances include the following information:
...
+1. Call out to the `resourcifier` service to instantiate the primitive resources
+described by the `expandedConfig`
 1. Respond with success or error messages to the original API request
 
-The manager is responsible for saving the metadata associated with
-deployments, manifests, type instances, and other resources in the Deployment
+`GET`, `PUT` and `DELETE` operations are processed in a similar manner, except
+that:
+
+* No expansion is performed for `GET` or `DELTE`
```
Comment (vaikas-google) [inline docs/design/design.md]: s/DELTE/DELETE/

**Comment by vaikas-google** [comment]
LGTM

**Comment by jackgr** [comment]
This pull request is ready to merge, but should not be merged until #77 is also ready to merge.

File: README.md [hunk 2]
```diff
@@ -2,27 +2,28 @@
...
-that second example correctly. It uses DM to deploy itself.
-See [examples/bootstrap/README.md](examples/bootstrap/README.md) for more information.)
+A configuration is just a `YAML` file that supplies parameters. (Yes, 
+you're reading that second example correctly. It uses DM to deploy itself. See
+[examples/bootstrap/README.md](examples/bootstrap/README.md) for more information.)
 
 DM runs server side, in your Kubernetes cluster, so it can tell you what types
-you've instantiated there, what instances you've created of a given type, and even
-how the instances are organized. So, you can ask questions like:
+you've instantiated there, including both primitive types and templates, what
```
Comment (bmelville) [inline README.md]: are we calling these primitive types or resource types?

**Comment by bmelville** [comment]
LGTM

**Pull request merged**